Download error not exposed with `HOMEBREW_DOWNLOAD_CONCURRENCY`

### `brew doctor` output

```shell
N/A - Output is from CI
```

### Verification

- [x] I ran `brew update` twice and am still able to reproduce my issue.
- [x] My "`brew doctor` output" above says `Your system is ready to brew` or a definitely unrelated `Tier` message.
- [x] This issue's title and/or description do not reference a single formula e.g. `brew install wget`. If they do, open an issue at https://github.com/Homebrew/homebrew-core/issues/new/choose instead.

### `brew config` output

```shell
HOMEBREW_VERSION: 5.0.11-6-g277c06d
ORIGIN: https://github.com/Homebrew/brew
HEAD: 277c06d55624e3056353b2b90141afd08251ac2f
Last commit: 67 minutes ago
Branch: main
Core tap HEAD: a57f2598342348ac66094c1b646fb823fa9cbd8c
Core tap last commit: 19 minutes ago
Core tap JSON: 19 Jan 09:44 UTC
Core cask tap HEAD: c8d004a06774e0502a2e3f07f9726628ea48332c
Core cask tap last commit: 3 minutes ago
Core cask tap branch: HEAD
Core cask tap JSON: 19 Jan 09:44 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: ["--no-quarantine"]
HOMEBREW_COLOR: set
HOMEBREW_DEVELOPER: set
HOMEBREW_DOWNLOAD_CONCURRENCY: 6
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 3
HOMEBREW_NO_AUTO_UPDATE: set
HOMEBREW_NO_INSTALL_CLEANUP: set
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.4.8 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.4.8/bin/ruby
CPU: 3-core 64-bit dunno
Clang: 17.0.0 build 1700
Git: 2.50.1 => /Applications/Xcode_26.0.1.app/Contents/Developer/usr/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 26.0.1-arm64
CLT: 26.2.0.0.1.1764812424
Xcode: 26.0.1 => /Applications/Xcode_26.0.1.app/Contents/Developer
Metal Toolchain: 17.0 (17A324)
Rosetta 2: false
```

### What were you trying to do (and why)?

When `brew fetch` fails the relevant error is no longer exposed, this is the case even with `--verbose --debug` set.

### What happened (include all command output)?

Current output:

```
Run brew fetch --cask --retry --force  './Casks/m/mudlet.rb'
 ✘ Cask mudlet (4.19.1)
Error: Download failed on Cask 'mudlet' with message: Download failed: https://www.mudlet.org/wp-content/files/Mudlet-4.19.1-arm64.dmg
Error: Process completed with exit code 1.
```

with `HOMEBREW_DOWNLOAD_CONCURRENCY=1` set

```
Run brew fetch --cask --retry --force  './Casks/m/mudlet.rb'
==> Downloading https://www.mudlet.org/wp-content/files/Mudlet-4.19.1-arm64.dmg
curl: (56) The requested URL returned error: 403
Error: Download failed on Cask 'mudlet' with message: Download failed: https://www.mudlet.org/wp-content/files/Mudlet-4.19.1-arm64.dmg
```


For the context of CI we *could* set HOMEBREW_DOWNLOAD_CONCURRENCY=1 for `homebrew-cask` because it only ever downloads one item in the fetch step, but I believe this variable may not be available forever. 

### What did you expect to happen?

That the relevant curl/network error would be visible.

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
With a know failing `url` in a cask;
run `brew fetch -vd <token>` and the error is suppressed.
run `HOMEBREW_DOWNLOAD_CONCURRENCY=1 brew fetch -vd <token>` and the error will be shown.
```

--- a/Library/Homebrew/download_queue.rb
+++ b/Library/Homebrew/download_queue.rb
@@ -110,8 +110,7 @@ module Homebrew
                 raise exception
               else
                 message = if exception.is_a?(DownloadError) && exception.cause.is_a?(ErrorDuringExecution)
-                  stderr_output = exception.cause.stderr
-                  if stderr_output.present?
+                  if (stderr_output = exception.cause.stderr.presence)
                     "#{stderr_output}#{exception.cause.message}"
                   else
                     exception.cause.message

